### PR TITLE
Update golangci-lint and config

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind ci

#### What this PR does / why we need it:
Updating the linter and enable further checks. No reports to fix for this one.
#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -57,7 +57,7 @@ integration-static: .install.ginkgo # It needs to be release so we correctly tes
 
 .install.golangci-lint:
 	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | \
-		BINDIR=$(abspath $(GOTOOLS_BINDIR)) sh -s v1.49.0
+		BINDIR=$(abspath $(GOTOOLS_BINDIR)) sh -s v1.50.1
```
Comment (rphillips) [inline Makefile:60]: nit: could you make the v1.50.1 a constant at the top of the Makefile?

Comment (saschagrunert) [reply Makefile]: Yep, done :+1:

**Comment by rphillips** [comment]
Thanks!

**Pull request merged**